admin-manual/maintenance/maintenance.rst

```diff
@@ -134,72 +142,287 @@ The index names are: `aips`, `aipfiles`, `transfers` and `transferfiles`.
+  rsync -av /var/archivematica/storage-service/storage.db /backup/location/storage.db
+
+.. note::
+
+  The Storage Service must not be actively in use. Make sure the
+  Storage Service is not running by stopping the ``nginx`` or ``storage-service``
+  services or by making the backup at a time that it is not in use.
+
+To restore Storage Service from backup:
+
+.. code:: bash
+
+  service archivematica-storage-service stop
+  rsync -av /backup/location/storage.db /var/archivematica/storage-service/storage.db
+  rsync -av /backup/location/storage_service /var/archivematica/storage_service
+  service archivematica-storage-service start
+
+Elasticsearch
+^^^^^^^^^^^^^
 
 Elasticsearch is used to store long-term data. Instructions and scripts for
 backing up and restoring Elasticsearch are available in the
 `Elasticsearch documentation`_.
 
+**Preconfiguration**
+
+The path.repo and snapshot repository have to be configured. For example, using
+``/var/lib/elasticsearch/backup-repo`` as the repo path:
+
+.. code:: bash
+
+  mkdir /var/lib/elasticsearch/backup-repo
+  chmod 0755 /var/lib/elasticsearch/backup-repo
+  chown elasticsearch:elasticsearch /var/lib/elasticsearch/backup-repo
+
+Add this line to the ``/etc/elasticsearch/elasticsearch.yml`` file:
+
+.. code:: bash
+
+  path.repo: /var/lib/elasticsearch/backup-repo
+
+Restart elasticsearch: 
+
+.. code:: bash
+
+  service elasticsearch restart
+
+To use a new directory as snapshot repository, create and adjust permissions for one, like so:
+
+.. code:: bash
+
+  mkdir /var/lib/elasticsearch/backup-repo/es_backup_clientname
+  chmod 0755 /var/lib/elasticsearch/backup-repo/es_backup_clientname
+  chown elasticsearch:elasticsearch /var/lib/elasticsearch/backup-repo/es_backup_clientname
+
+Before any snapshot or restore operation can be performed, a snapshot repository
+should be registered in Elasticsearch. The repository settings are
+repository-type specific:
+
+.. code:: bash
+
+  curl -XPUT 'http://localhost:9200/_snapshot/es_backup_clientname' -d '{
+      "type": "fs",
+      "settings": {
+          "compress" : true,
+          "location": "/var/lib/elasticsearch/backup-repo/es_backup_clientname"
+      }
+  }'
+
+**Backing up Elasticsearch indexes**
+
+To make a backup (snapshot) for the ``aips`` and ``transfer`` indexes, a different name
+has to be used every time a snapshot is taken. For example, using the date
+inside the filename:
+
+.. code:: bash
+
+  curl -XPUT "http://localhost:9200/_snapshot/es_backup_clientname/snapshot_aips_20170726?pretty?wait_for_completion=true" -d'
+  {
+    "indices": "aips",  
+    "ignore_unavailable": true,
+    "include_global_state": false
+  }'
+
+  curl -XPUT "http://localhost:9200/_snapshot/es_backup_clientname/snapshot_transfers_20170726?pretty?wait_for_completion=true" -d'
+  {
+    "indices": "transfers",  
+    "ignore_unavailable": true,
+    "include_global_state": false
+  }'
```

Comment:
You could now do all the indexes in one run and use date match expressions to auto-generate the date:

```
curl -XPUT "http://localhost:9200/_snapshot/es_backup_clientname/%3Csnapshot-am-%7Bnow%2Fd%7D%3E?wait_for_completion=true" -d'
{
  "indices": "aips,aipfiles,transfers,transferfiles",
  "ignore_unavailable": true,
  "include_global_state": false
}'
```